Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -422,6 +422,25 @@ class HTTPClientTests: XCTestCase {
         }
     }
 
+    func testConnectTimeout() throws {
+        let httpClient = HTTPClient(eventLoopGroupProvider: .shared(self.clientGroup),
+                                    configuration: .init(timeout: .init(connect: .milliseconds(100), read: .milliseconds(150))))
+
+        defer {
+            XCTAssertNoThrow(try httpClient.syncShutdown())
+        }
+
+        // This must throw as 198.51.100.254 is reserved for documentation only
+        XCTAssertThrowsError(try httpClient.get(url: "http://198.51.100.254:65535/get").wait()) { error in
```

Comment:
@weissi I looked this up here:
https://github.com/apple/swift-nio/blob/6046aaa6291fb075c83b5c22bad41b9780b2d977/Tests/NIOTests/ChannelTests.swift#L1060
is this ok? or is there a better way to get a connect timeout?

Reply:
This should be just fine. Misconfigured networks can make this test fail, but without a complete user space IP and TCP implementation we don't have a better option.